### ISSUE TITLE
Lingering close doesn't terminate before timeout

### ISSUE THREAD

**Issue by heftig** [aio-libs/aiohttp#1559]
## Long story short

My web app is occasionally freezing for 30s, burning CPU all the while.

## Steps to reproduce

Not clear. There's a lot of code involved.

The problem disappears when:
- I set `lingering_time=0`
- I change line 277 in `aiohttp/server.py` from
```python
while self._loop.time() < end_time:
```
to
```python
while not payload.is_eof() and self._loop.time() < end_time:
```
- I use `loop.set_debug(True)` to enable async debugging. There are no more uncompleted requests, either. Maybe the protocol gets slowed down enough to receive the EOF early.

## Your environment

aiohttp 1.2.0, python 3.6, Linux

**Comment by fafhrd91** [comment]
good catch! thanks!

**Comment by fafhrd91** [comment]
fixed in master 4a59c4f4cbb525fda591eaa49ebcc8723a0a0da6
